app/views/books/new.html.erb

```diff
@@ -5,6 +5,18 @@
   <h1 class="title">New Book Form</h1>
 
   <%= form_for(@book) do |f| %>
+
+  <% if @book.errors.any? %>
+    <div id="error_explanation">
```

Comment:
Use the standard flash message pattern in your controller instead, and get rid of this block of code (e.g., 
```
if @book.save 
  flash[:success]
```